ts/Stock/Indicators/TrendLine/TrendLineIndicator.ts

```diff
@@ -107,48 +107,55 @@ class TrendLineIndicator extends SMAIndicator {
         series: TLinkedSeries,
         params: TrendLineParamsOptions
     ): IndicatorValuesObject<TLinkedSeries> {
-        const xVal: Array<number> = (series.xData as any),
+        const orgXVal: Array<number> = (series.xData as any),
             yVal: Array<Array<number>> = (series.yData as any),
+            xVal: Array<number> = [],
             LR: Array<Array<number>> = [],
             xData: Array<number> = [],
             yData: Array<number> = [],
-            xValLength: number = xVal.length,
-            index: number = (params.index as any);
-
-        let sumX = (xValLength - 1) * xValLength / 2,
-            sumY = 0,
-            sumXY = 0,
-            sumX2 =
-                ((xValLength - 1) * (xValLength) * (2 * xValLength - 1)) / 6,
-            alpha: number,
-            i: number,
-            y: number;
-
-        // Get sums:
-        for (i = 0; i < xValLength; i++) {
-            y = isArray(yVal[i]) ? yVal[i][index] : (yVal[i] as any);
-            sumY += y;
-            sumXY += i * y;
+            index: number = params.index || 3;
```

Comment:
No need for that. If index can be undefined, then it means TS is incorrect (eg DeepPartial is not used)

Reply:
@pawelfus What would you say about the solution in [Added index to TrendLine interface.](https://github.com/highcharts/highcharts/pull/19952/commits/83f26f7fa6ba6cbc1a0c35bf8f10cd0b7b596832) commit?

Reply:
Looks good!